packages/dds/sequence-sequential-id/src/sharedStringWithSequentialId.ts

```diff
@@ -0,0 +1,88 @@
+                    return true;
+                });
+                return set;
+            });
+            this.on("maintenance", this.applyIdToLocalAckedSegment);
+            this.on("sequenceDelta", this.applyIdToRemoteAckedSegment);
+        }
+
+        public static getFactory() {
+            return new SharedStringWithSequentialIdFactory();
+        }
+
+        private readonly applyIdToRemoteAckedSegment = (event: SequenceDeltaEvent): void => {
+            if (event.isLocal) {
+                // Do not apply id for local changes
+                return;
+            }
+
+            event.ranges.forEach((range) => {
+            const markerSegment = range.segment;
+                if (Marker.is(markerSegment)) {
+                    if (event.deltaOperation === MergeTreeDeltaType.INSERT) {
+                        this.applyIdToMarker(markerSegment);
+                    } else if (event.deltaOperation === MergeTreeDeltaType.REMOVE) {
+                        this.sortedMarkers.value.remove(markerSegment);
+                    }
+                }
+            });
+        };
+
+        private readonly applyIdToLocalAckedSegment = (event: SequenceMaintenanceEvent): void => {
+            if (event.deltaArgs.operation === MergeTreeMaintenanceType.ACKNOWLEDGED) {
+                event.ranges.forEach((range) => {
+                    const markerSegment = range.segment;
+                    if (Marker.is(markerSegment)) {
+                        if (event.opArgs?.op.type === MergeTreeDeltaType.INSERT) {
+                            this.applyIdToMarker(markerSegment);
+                        } else if (event.opArgs?.op.type === MergeTreeDeltaType.REMOVE) {
+                            this.sortedMarkers.value.remove(markerSegment);
+                        }
+                    }
+                });
+            }
+        };
+
+        private readonly applyIdToMarker = (markerSegment: Marker): void => {
+            this.sortedMarkers.value.addOrUpdate(markerSegment);
+            const markerItems = this.sortedMarkers.value.items;
+            const newMarkerIndex = markerItems.indexOf(markerSegment);
+            const previousMarkerIndex = newMarkerIndex - 1;
+            const nextMarkerIndex = newMarkerIndex + 1;
+            const previousMarker = previousMarkerIndex >= 0 ?
+             markerItems[previousMarkerIndex] as Marker : undefined;
+            const nextMarker = nextMarkerIndex < markerItems.length ?
+             markerItems[nextMarkerIndex] as Marker : undefined;
+            const previousId: string = previousMarker !== undefined ? previousMarker.getId() ?? digitZero : digitZero;
+            const nextId: string = nextMarker !== undefined ? nextMarker.getId() ?? digitLast : digitLast;
+            let newMarkerProps = markerSegment.properties ?? {};
+            const id = createIdBetween(previousId, nextId);
+            newMarkerProps = { ...newMarkerProps, [reservedMarkerIdKey]: id };
```

Comment:
Not sure, do we need to copy properties object? I.e. what happens if I just stuck a new property on existing object?

Reply:
remove copy of properties and put new property into existing propertyset